core/src/main/java/com/twosigma/beaker/core/module/config/DefaultBeakerConfig.java

```diff
@@ -94,7 +96,18 @@ public DefaultBeakerConfig(BeakerConfigPref pref, GeneralUtils utils)
     this.useKerberos = pref.getUseKerberos();
     this.portBase = pref.getPortBase();
     this.reservedPortCount = 4;
-    this.dotDir = System.getProperty("user.home") + "/.beaker/v1";
+
+
+    if (pref.getPortable()){
+      String path = Paths.get("").toAbsolutePath().toString();
+      boolean endsWithSlash = path.endsWith(File.separator);
+      path = path.substring(0,
```

Comment:
what does this do?  is this removing the last directory?
if so then better to use subpath and getNameCount?
we should not do so much detailed string manipulation of filenames.